graph/collection-properties.md

```diff
@@ -0,0 +1,402 @@
+
+```HTTP
+GET  /interestingData/bars
+
+200 OK
+{
+  "value": [
+    {
+      "id": "firstBarId",
+      "differentProperty": 10
+    },
+    {
+      "id": "secondBarId",
+      "differentProperty": -6914
+    },
+    {
+      "id": "thirdBarId",
+      "differentProperty": 42
+    }
+  ]
+}
+```
+
+## Retrieving individual elements from a collection
+
+The [OData standard](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_RequestingIndividualEntities) specifies that clients can retrieve individual elements of a collection of entity types using a `GET` request:
+> To retrieve an individual entity, the client makes a GET request to a URL that identifies the entity, e.g. its read URL.
+
+There is no way to do this for complex types because there is no way to address a particular instance of a complex type within a collection. 
+
+### Entity Type
+
+```HTTP
+GET  /interestingData/bars/firstBarId
+
+200 OK
+{
+  "id": "firstBarId",
+  "differentProperty": 10
+}
+```
+
+## Removing individual elements from a collection
+
+The [OData standard](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_DeleteanEntity) specifies that clients can remove individual elements of a collection of entity types using a `DELETE` request:
+> To delete an individual entity, the client makes a DELETE request to a URL that identifies the entity.
+
+There is no way to do this for complex types because there is no way to address a particular instance of a complex type within a collection. 
+
+### Entity Type
+
+#### Remove an element from the collection
+
+```HTTP
+DELETE  /interestingData/bars/thirdBarId
+
+204 No Content
+```
+
+#### Check the new contents of the collection
+
+```HTTP
+GET  /interestingData/bars
+
+200 OK
+{
+  "value": [
+    {
+      "id": "firstBarId",
+      "differentProperty": 10
+    },
+    {
+      "id": "secondBarId",
+      "differentProperty": -6914
+    }
+  ]
+}
+```
+
+## Updating individual elements in a collection
+
+The [OData standard](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_UpdateanEntity) specifies that clients can update individual elements of a collection of entity types using a `PATCH` request:
+> To update an individual entity, the client makes a PATCH or PUT request to a URL that identifies the entity.
+
+There is no way to do this for complex types because there is no way to address a particular instance of a complex type within a collection. 
+
+### Entity Type
+
+#### Update an element in the collection
+
+```HTTP
+PATCH  /interestingData/bars/firstBarId
+{
+  "differentProperty": "15"
+}
+
+204 No Content
+```
+
+#### Check the new contents of the collection
+
+```HTTP
+GET  /interestingData/bars
+
+200 OK
+{
+  "value": [
+    {
+      "id": "firstBarId",
+      "differentProperty": 15
+    },
+    {
+      "id": "secondBarId",
+      "differentProperty": -6914
+    }
+  ]
+}
+```
+
```

Comment:
also, you can have other objects reference an entity within a collection, which you cannot do with a complex type:

```xml
<EntityType Name="baz">
      <Property Name="Description" Type="Edm.String"/>
      <NaviationProperty Name="bar" Type="self.bar"/>
</EntityType>
```

```HTTP
GET interestingData/baz
200 OK
{
    "Description": "my favorite bar",
    "bar": 
    {
        "id": "firstBarId",
        "differentProperty": 15
     }
}
```